tests/router/test_router_e2e_with_vllm.py

```
@@ -434,21 +455,58 @@ async def launch_workers_with_indexer(self, endpoint):
+                    # diagnostic is guarded so it can never mask the timeout it describes.
+                    try:
+                        returncode = process.proc.poll() if process.proc else None
+                        if process.proc is None:
+                            liveness = "subprocess was never started"
+                        elif returncode is None:
+                            liveness = "subprocess still running"
+                        else:
+                            liveness = (
+                                f"subprocess already exited with code {returncode}"
+                            )
+                    except Exception as diag_exc:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)